imap_processing/lo/l0/compression_tables.py

```diff
@@ -0,0 +1,853 @@
+    "1010": 10,
+    "1011": 11,
+    "1100": 12,
+    "1101": 13,
+}
+
+tof_decoder_table = {
+    # First level of keys are the case numbers
+    0: {
+        1: {
+            "ENERGY": 3,
+            "POS": 0,
+            "TOF0": 10,
+            "TOF1": 0,
+            "TOF2": 9,
+            "TOF3": 6,
+            "CKSM": 3,
+            "TIME": 12,
+        },
+        0: {
+            "ENERGY": 3,
+            "POS": 0,
+            "TOF0": 10,
+            "TOF1": 9,
+            "TOF2": 9,
+            "TOF3": 6,
+            "CKSM": 0,
+            "TIME": 12,
+        },
+    },
+    1: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 10,
+        "TOF1": 9,
+        "TOF2": 9,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    2: {
+        "ENERGY": 3,
+        "POS": 2,
+        "TOF0": 9,
+        "TOF1": 9,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    3: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 11,
+        "TOF1": 0,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    4: {
+        "ENERGY": 3,
+        "POS": 2,
+        "TOF0": 10,
+        "TOF1": 0,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    5: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 11,
+        "TOF1": 0,
+        "TOF2": 9,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    6: {
+        "ENERGY": 3,
+        "POS": 2,
+        "TOF0": 10,
+        "TOF1": 0,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    7: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 11,
+        "TOF1": 0,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    8: {
+        "ENERGY": 3,
+        "POS": 2,
+        "TOF0": 0,
+        "TOF1": 9,
+        "TOF2": 9,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    9: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 0,
+        "TOF1": 10,
+        "TOF2": 10,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    10: {
+        "ENERGY": 3,
+        "POS": 2,
+        "TOF0": 0,
+        "TOF1": 10,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    11: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 0,
+        "TOF1": 11,
+        "TOF2": 0,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    12: {
+        "ENERGY": 3,
+        "POS": 2,
+        "TOF0": 0,
+        "TOF1": 0,
+        "TOF2": 10,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+    13: {
+        "ENERGY": 3,
+        "POS": 0,
+        "TOF0": 0,
+        "TOF1": 0,
+        "TOF2": 11,
+        "TOF3": 0,
+        "CKSM": 0,
+        "TIME": 12,
+    },
+}
+
+tof_calculation_table = {
```

Comment:
In general, I feel like tables should be lists that can be indexed into rather than dicts since you're just adding the explicit ints here but needing a hash call. If you like to have the identifier perhaps it could be a comment `# item 0`

```python
tof_calculation_table = [{"TIME": "0xFFF", ...}, {"ITEM1"...}, ...]
```

edit: I see you did this to the one right below this already, so there is precedence :)

Reply:
I have the dictionary with the field names so it's easier to keep track of which fields should be missing for which cases. Maxine had another comment suggesting that this could be a class, so I might try that approach. That might let me be explicit about field names, while removing the dictionary.